logstash-core/src/main/java/org/logstash/instrument/witness/PipelinesWitness.java

```diff
@@ -0,0 +1,35 @@
+package org.logstash.instrument.witness;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Witness for the set of pipelines.
+ */
+final public class PipelinesWitness {
+
+    private final Map<String, PipelineWitness> pipelines;
+
+    /**
+     * Constructor.
+     */
+    public PipelinesWitness() {
+        this.pipelines = new ConcurrentHashMap<>();
+    }
+
+    /**
+     * Get a uniquely named pipeline witness. If one does not exist, it will be created.
+     *
+     * @param name The name of the pipeline.
+     * @return the {@link PipelineWitness} identified by the given name.
+     */
+    public PipelineWitness pipeline(String name) {
+        if (pipelines.containsKey(name)) {
```

Comment:
Nice catch, you are right... if multiple threads try adding the same named pipeline at the same time, the last one would win wiping out any data from the prior.  

I will ensure it is a ConcurrentHashMap and use the `computeIfAbsent` method.